src/heap.py

```diff
@@ -43,15 +45,17 @@ def pop(self):
         try:
             return_val = self.heap[0]
             del self.heap[0]
-            self.heap.insert(0, self.heap.pop(len(self.heap) - 1))
-            # self._sift_down()
+            self.heap.insert(0, self.heap.pop(self.length() - 1))
+            self._sift_down()
             return return_val
         except IndexError:
             print('This heap is empty.')
 
+    def length(self):
+        return len(self.heap)
 
-new_heap = Heap([34, 56, 78, 90, 3, 4, 1, 6, 29, 66, 5, 4, 8, 9, 24, 20,
+
+new_heap = Heap([10, 7, 8, 3, 1, 4, 2, 12])
+big_heap = Heap([34, 56, 78, 90, 3, 4, 1, 6, 29, 66, 5, 4, 8, 9, 24, 20,
                  24, 45, 92, 93, 65, 87, 19, 9, 46, 72, 85, 4, 76, 900])
-print(new_heap.heap)
-new_heap.pop()
-print(new_heap.heap)
+print(big_heap.pop())
```

Comment:
Adding these test cases in the file should be done in an `if __name__ == '__main__':` block.